tests_integration/test_snowpark_function.py

```diff
@@ -0,0 +1,417 @@
+    )
+    _test_steps.assert_that_only_these_files_are_staged_in_snowflake(
+        f"deployments/{function_name}/app.zip"
+    )
+
+    _test_steps.snowpark_function_list_should_return_list_containing_function_at_the_first_place(
+        function_name=function_name,
+        arguments="()",
+        result_type="NUMBER",
+    )
+
+    _test_steps.snowpark_function_execute_should_return_expected_value(
+        function_name=function_name,
+        arguments="()",
+        expected_value=1,
+    )
+
+    _test_steps.snowpark_function_drop_should_finish_successfully(
+        function_name=function_name,
+        arguments="()",
+    )
+    _test_steps.assert_that_no_functions_are_in_snowflake()
+    # I'm not sure if it is expected behavior to leave staged files belonging to the removed function.
+    _test_steps.assert_that_only_these_files_are_staged_in_snowflake(
+        f"deployments/{function_name}/app.zip"
+    )
+
+    _test_steps.snowpark_function_list_should_return_no_data()
+
+
+@pytest.fixture
+def _test_setup(
+    runner,
+    snowflake_session,
+    sql_test_helper,
+    object_name_provider,
+    temporary_working_directory,
+    snapshot,
+):
+    snowpark_function_test_setup = SnowparkFunctionTestSetup(
+        runner=runner,
+        snowflake_session=snowflake_session,
+        sql_test_helper=sql_test_helper,
+        object_name_provider=object_name_provider,
+        snapshot=snapshot,
+    )
+    yield snowpark_function_test_setup
+    snowpark_function_test_setup.clean_after_test_case()
+
+
+@pytest.fixture
+def _test_steps(_test_setup):
+    yield SnowparkFunctionTestSteps(_test_setup)
+
+
+class SnowparkFunctionTestSetup:
+    def __init__(
+        self,
+        runner: SnowCLIRunner,
+        snowflake_session: SnowflakeConnection,
+        sql_test_helper: SqlTestHelper,
+        object_name_provider: ObjectNameProvider,
+        snapshot: SnapshotAssertion,
+    ):
+        self.runner = runner
+        self.snowflake_session = snowflake_session
+        self.sql_test_helper = sql_test_helper
+        self.test_object_name_provider = object_name_provider
+        self.object_name_prefix = object_name_provider.get_object_name_prefix()
+        self.snapshot = snapshot
+
+    def query_functions_created_in_this_test_case(self) -> List[Dict[str, Any]]:
+        results = self.sql_test_helper.execute_single_sql(
+            f"SHOW USER FUNCTIONS LIKE '{self.object_name_prefix.upper()}%'"
+        )
+        return results
+
+    def query_stages_in_database(self) -> List[Dict[str, Any]]:
+        return self.sql_test_helper.execute_single_sql(
+            f"SHOW STAGES IN DATABASE {self.snowflake_session.database}"
+        )
+
+    def query_files_uploaded_in_this_test_case(self) -> List[Dict[str, Any]]:
+        stages_in_database = [
+            stage["name"].lower() for stage in self.query_stages_in_database()
+        ]
+        if "deployments" in stages_in_database:
```

Comment:
I hope other commands don't use `deployments` stage 🙏

Reply:
I'm not sure I understand this question. What if they use the same stage?